src/app/featuredToursPage/[tourId]/[displayId]/page.tsx

```diff
@@ -1,74 +1,131 @@
+import NavBar from '../../../../components/userComponents/navBar/navBar';
+import { fetchDisplay } from '../../../../supabase/displays/queries';
+import { fetchTourDisplays } from '../../../../supabase/tour_displays/queries';
 
-export default function Page({ params }: { params: { displayId: string } }) {
+/**
+ * The page that displays a tour stop.
+ * @param params -
+ * @param params.params -
+ * @param params.params.tourId - The tour ID.
+ * @param params.params.displayId - The display ID.
+ * @returns The tour stop page.
+ */
+export default function TourStopPage({
+  params,
+}: {
+  params: { tourId: string; displayId: string };
+}) {
   const [display, setDisplay] = useState<DisplayRow>();
+  const [prev, setPrev] = useState<string>(
+    `/featuredToursPage/${params.tourId}`,
+  );
+  const [next, setNext] = useState<string>(
+    `/featuredToursPage/${params.tourId}/tourEndPage`,
+  );
 
   useEffect(() => {
-    async function fetchDisplay() {
-      try {
-        const { data, error } = await supabase
-          .from('displays')
-          .select('*')
-          .eq('id', params.displayId)
-          .single();
-        if (error) {
-          throw error;
-        }
-        if (!data) {
-          throw new Error('No data found');
-        }
-        console.log('Obtained display details');
-        const responseData: DisplayRow = data;
-        setDisplay(responseData);
-      } catch (error) {
-        console.error('Error fetching tour details:', error);
+    // Get display
+    const getDisplay = async () => {
+      const fetchedDisplay = await fetchDisplay(params.displayId);
+      setDisplay(fetchedDisplay);
+    };
+
+    // Get tour displays
+    const getTourDisplays = async () => {
+      const fetchedTourDisplays = await fetchTourDisplays(params.tourId);
+      return fetchedTourDisplays;
+    };
+
+    // Get the links for the previous and next pages
+    /**
+     *
+     */
+    async function getLinks() {
+      const tourDisplays: TourDisplaysRow[] = await getTourDisplays();
+      const index = tourDisplays.findIndex(
+        tourDisplay => tourDisplay.display_id === params.displayId,
+      );
+
+      if (index === -1) {
+        throw new Error('Display not found in tour displays');
+      } else if (index === 0) {
+        setNext(
+          `/featuredToursPage/${params.tourId}/${
+            tourDisplays[index + 1].display_id
+          }`,
+        );
+      } else if (index === tourDisplays.length - 1) {
+        setPrev(
+          `/featuredToursPage/${params.tourId}/${
+            tourDisplays[index - 1].display_id
+          }`,
+        );
+      } else {
+        setPrev(
+          `/featuredToursPage/${params.tourId}/${
+            tourDisplays[index - 1].display_id
+          }`,
+        );
+        setNext(
+          `/featuredToursPage/${params.tourId}/${
+            tourDisplays[index + 1].display_id
+          }`,
+        );
       }
```

Comment:
For readability + conciseness:  You can probably use a ternary to clean up all of the boolean logic.
Example below (haven't tested but something like this would work, you can probably make it more efficient too)
I prefer Option 2 
```suggestion
    OPTION 1:
    
      } else {
       let prevStop = index === 0 ? `/featuredToursPage/${params.tourId}`
        : `/featuredToursPage/${params.tourId}/${tourDisplays[index - 1].display_id
          }`
       let nextStop = index == tourDisplays.length - 1 ? `/featuredToursPage/${params.tourId}/tourEndPage` 
       : `/featuredToursPage/${params.tourId}/${
            tourDisplays[index + 1].display_id
          }`
          setPrev(prevStop)
          setNext(nextStop)
         
         OPTION 2
         OR you could try:
         let prevStop = index === 0 ? prev  : prev + `/${tourDisplays[index - 1].display_id}`
         let nextStop = index === tours.display.length - 1 ? next : `/featuredToursPage/${params.tourId}/${tourDisplays[index + 1].display_id}`
         setPrev(prevStop)
         setNext(nextStop)
       
```